File: pyworkspace/packages/a2a-sdk/src/trogonai_a2a_sdk/interfaces.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Mapping, Optional, Protocol, runtime_checkable

from .types import AgentId, AgentRecord, ExchangeRequest, ExchangeResponse


@runtime_checkable
class SvidSource(Protocol):
    async def current(self) -> str: ...


@runtime_checkable
class SubjectTokenSource(Protocol):
    async def current(self) -> str: ...


@runtime_checkable
class Sts(Protocol):
    async def exchange(self, req: ExchangeRequest) -> ExchangeResponse: ...


@runtime_checkable
class Registry(Protocol):
    async def lookup(self, agent_id: AgentId) -> AgentRecord: ...


@runtime_checkable
class MessageTransport(Protocol):
    async def request(
        self,
        subject: str,
        payload: bytes,
        headers: Mapping[str, str],
    ) -> bytes: ...


@dataclass(frozen=True)
class JwksKey:
    key: str
    kid: Optional[str] = None


@runtime_checkable
class Jwks(Protocol):
    async def decoding_key(self, kid: Optional[str] = None) -> JwksKey: ...

File: tsworkspace/packages/a2a-sdk/src/interfaces.ts
import type {
  AgentId,
  AgentRecord,
  ExchangeRequest,
  ExchangeResponse,
} from "./types.js";

export interface SvidSource {
  current(): Promise<string>;
}

export interface SubjectTokenSource {
  current(): Promise<string>;
}

export interface Sts {
  exchange(req: ExchangeRequest): Promise<ExchangeResponse>;
}

export interface Registry {
  lookup(agentId: AgentId): Promise<AgentRecord>;
}

export interface MessageTransport {
  request(
    subject: string,
    payload: Uint8Array,
    headers: Record<string, string>,
  ): Promise<Uint8Array>;
}

export interface JwksKey {
  kid?: string;
  key: string;
}

export interface Jwks {
  decodingKey(kid?: string): Promise<JwksKey>;
}
